src/handler.ts

```diff
@@ -1,94 +1,807 @@
+        dag_transactions: { select: { hash: true } },
+        global_snapshots: true,
+        super: { include: { block_parents: true } }
+      }
+    });
+    return respond(block, dagBlockResponse);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+const transactionsQuery = async (
+  where,
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const query = {
+      ...where,
+      include: {
+        dag_blocks: {
+          include: {
+            global_snapshots: { select: { hash: true, ordinal: true } }
+          }
+        }
+      },
+      orderBy: { ordinal: 'desc' }
+    };
+
+    const toCursor = (row) => ({
+      ...toCreatedAtOrdinalCursor(row),
+      hash: row.hash
+    });
+
+    const fromCursor = (row) => ({
+      ...fromCreatedAtOrdinalCursor(row),
+      hash: row.hash
+    });
+
+    return await paginatedQuery(
+    extractPagination(event),
+      toCursor,
+      fromCursor,
+      query,
+      prisma.dag_transactions.findMany,
+      dagTransactionsResponse
+    );
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const transactions = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  return transactionsQuery({}, event);
+};
+
+export const transaction = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { hash } = event.pathParameters || {};
+
+    const transaction = await prisma.dag_transactions.findUnique({
+      where: { hash },
+      include: {
+        dag_blocks: {
+          include: {
+            global_snapshots: { select: { hash: true, ordinal: true } }
+          }
+        }
+      }
+    });
+
+    return respond(transaction, dagTransactionResponse);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const transactionsByAddress = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { address } = event.pathParameters || {};
+
+    const where = {
+      where: { OR: [{ source_addr: address }, { destination_addr: address }] }
+    };
+
+    return transactionsQuery(where, event);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const transactionsBySource = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { address } = event.pathParameters || {};
+
+    const where = { where: { source_addr: address } };
+
+    return transactionsQuery(where, event);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const transactionsByDestination = async (
+  event: APIGatewayProxyEvent
+): Promise<APIGatewayProxyResult> => {
+  try {
+    const { address } = event.pathParameters || {};
+
+    const where = { where: { destination_addr: address } };
+
+    return transactionsQuery(where, event);
+  } catch (error) {
+    return handleError(error);
+  }
+};
+
+export const balanceByAddress = async (
```

Comment:
Could you take another pass at a naming convention so this is easier to follow/understand? 

I would suggest `handleDAGBalanceByAddress` for this:
- Always call out DAG specifically
- "handle" prefix for all methods that are directly called out in handler.ts